portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -13,20 +14,72 @@
 // limitations under the License.
 
 package com.google.sps.servlets;
-
+import java.util.ArrayList;
 import java.io.IOException;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
+import com.google.gson.*;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.appengine.api.datastore.DatastoreFailureException;
 
-/** Servlet that returns some example content. TODO: modify this file to handle comments data */
+/** Servlet that handles comments */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
-  @Override
-  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello world!</h1>");
-  }
+    public static final String JSON_CONTENT_TYPE = "application/json";
+    public static final String COMMENT_PROPERTY = "comment";
+    public static final String TIMESTAMP_PROPERTY = "timestamp";
+    public static final String COMMENT_ENTITY = "comment";
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    private final Gson parser = new Gson();
+
+    @Override
+    public void doGet(HttpServletRequest req, HttpServletResponse res) throws IOException{
+        int limit = Integer.parseInt(req.getParameter("limit"));
+
+        Query query = new Query(COMMENT_PROPERTY).addSort(TIMESTAMP_PROPERTY, SortDirection.DESCENDING);
+        PreparedQuery results = datastore.prepare(query);
+        ArrayList comments = new ArrayList();
+        //Integer to keep track of how many comments have been added to the arraylist that will be returned
+        int count = 1;
+        for (Entity entity : results.asIterable()){
+            //Stops iteration after 'limit' iterations
+            if(count > limit) {
+               break; 
+            } 
+            //Add's the entity's comment string to the arraylist that will be returned
+            comments.add(entity.getProperty(COMMENT_PROPERTY));
+            ++count;
+        }
+        res.setContentType(JSON_CONTENT_TYPE);
+
+        res.getWriter().println(parser.toJson(comments));
+    }
+
+    @Override
+    public void doPost(HttpServletRequest req, HttpServletResponse res) throws IOException{
+        String jsonstring = req.getParameter("comment");
+        Entity commentEntity = new Entity(COMMENT_ENTITY);
+        commentEntity.setProperty(COMMENT_PROPERTY, jsonstring);
+
+        long timestamp = System.currentTimeMillis();
```

Comment:
this is a error prone. Change timestamp to be currentTimeMillis. You lost unit totally after this line.

Reply:
I'm not really understanding what the issue with the code is. Could you please rephrase?